src/functions/instance/getTableData.js

```diff
@@ -2,6 +2,20 @@ import describeTable from '../api/instance/describeTable';
 import searchByValue from '../api/instance/searchByValue';
 import searchByConditions from '../api/instance/searchByConditions';
 
+const getAttributesFromTableData = (tableData, existingAttributes) => {
+  const existing = new Set(existingAttributes);
+  const extra = new Set();
+  for (const dataRow of tableData) {
+    for (const key of Object.keys(dataRow)) {
```

Comment:
if  `extra` was a normal array, maybe you can do something like this: 
```javascript
const keys = new Set(Object.keys(dataRow));
extra.push(...Array.from(keys.intersection(existing)));
```

here are some set comparison functions: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#set_composition